neqo-transport/src/packet.rs

```diff
@@ -382,6 +391,7 @@ impl<'a> PublicPacket<'a> {
                         scid: None,
```

Comment:
Another thought occurred.  There are cases where we don't care what version the packet is (take the server case where we haven't decided on what version to use).  But in most cases we really do care.  We don't want to send a -27 Initial and allow the server to send a -28 Initial back in response.

I think that this function should take an `Option<Version>` so that it can decode with a fixed assumption about version.  It's only the server decoding Initial packets that really needs to worry about handling multiple versions.  I suppose also if we ever get into compatible version negotiation we might need to move to `Vec<Version>` instead, but we can deal with that then.

Reply:
I don't think this is a problem -- we get the quic version from the initial and then create the matching version of server connection, assuming we get that far.

I did notice that our version negotiation packet should report support for both (plus grease), so I've made that change.